packages/cadl-ranch-specs/http/payload/xml/main.tsp

```diff
@@ -0,0 +1,195 @@
+import "@typespec/http";
+import "@azure-tools/cadl-ranch-expect";
+import "@azure-tools/typespec-client-generator-core";
+
+using TypeSpec.Http;
+using Azure.ClientGenerator.Core;
+
+@doc("Sends and receives bodies in XML format.")
+@scenarioService("/payload/xml")
+namespace Payload.Xml;
+
+@doc("Contains fields of primitive types.")
+model SimpleModel {
+  name: string;
+  age: int32;
+}
+
+@doc("Contains fields of arrays of primitive types.")
+model ModelWithSimpleArrays {
+  colors: string[];
+  counts: int32[];
+}
+
+@doc("Contains an array of models.")
+model ModelWithArrayOfModel {
+  items: SimpleModel[];
+}
+
+@doc("Contains an optional field.")
+model ModelWithOptionalField {
+  item: string;
+  counts?: int32;
+}
+
+@scenario
+@route("/simpleModel")
+@scenarioDoc("""
+Returns an instance of SimpleModel.
+Expected response body:
+```xml
+<SimpleModel>
+  <Name>foo</Name>
+  <Age>123</Age>
+</SimpleModel>
+```
+""")
+@get
+op getSimpleModel(): {
+  @header("content-type") contentType: "application/xml";
+  @body body: SimpleModel;
+};
+
+@scenario
+@route("/simpleModel")
+@scenarioDoc("""
+Sends an instance of SimpleModel.
+Expected request body:
+```xml
+<SimpleModel>
+  <Name>foo</Name>
+  <Age>123</Age>
```

Comment:
I don't this we should Pascale case the properties automatically. The spec doesn't say that

Reply:
I can change this, but I presume we need to support both cases (storage pascal-cases their XML responses). If the default expectation is to assume camel-case for properties, I can change this and add separate tests for pascal case.

Reply:
It's up to a service to judge what they want to do, either they use the pascal case name to start with or use the xml.name decorator/@encodedName to specify the right name over the wire

Reply:
Ok, so if I correctly understand the [guide](https://typespec.io/docs/next/libraries/xml/guide), the casing for properties should match the casing in the tsp. Is that what we're saying?

Reply:
yeah

Reply:
fixed